src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxTests.cs

```diff
@@ -384,6 +386,211 @@ public static void ReadDSAPrivateKey()
+                AssertExtensions.SequenceEqual(MLKemTestData.IncrementalSeed, kem.ExportPrivateSeed());
+            }
+        }
+
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [MemberData(nameof(StorageFlags))]
+        public static void ReadMLKem1024PrivateKey_ExpandedKey_Pfx(X509KeyStorageFlags keyStorageFlags)
+        {
+            byte[] pfxBytes = MLKemTestData.IetfMlKem1024PrivateKeyExpandedKeyPfx;
+            string pfxPassword = MLKemTestData.EncryptedPrivateKeyPassword;
+
+            using (X509Certificate2 cert = X509CertificateLoader.LoadPkcs12(pfxBytes, pfxPassword, keyStorageFlags))
+            using (MLKem kem = cert.GetMLKemPrivateKey())
+            {
+                Assert.NotNull(kem);
+                Assert.Equal(MLKemAlgorithm.MLKem1024, kem.Algorithm);
+                Assert.Equal("CN=LAMPS WG, O=IETF", cert.Subject);
+                Assert.Throws<CryptographicException>(() => kem.ExportPrivateSeed());
+                AssertExtensions.SequenceEqual(
+                    MLKemTestData.IetfMlKem1024PrivateKeyDecapsulationKey,
+                    kem.ExportDecapsulationKey());
+            }
+        }
+
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [MemberData(nameof(StorageFlags))]
+        public static void ReadMLKem1024PrivateKey_Both_Pfx(X509KeyStorageFlags keyStorageFlags)
+        {
+            byte[] pfxBytes = MLKemTestData.IetfMlKem1024PrivateKeyBothPfx;
+            string pfxPassword = MLKemTestData.EncryptedPrivateKeyPassword;
+
+            using (X509Certificate2 cert = X509CertificateLoader.LoadPkcs12(pfxBytes, pfxPassword, keyStorageFlags))
+            using (MLKem kem = cert.GetMLKemPrivateKey())
+            {
+                Assert.NotNull(kem);
+                Assert.Equal(MLKemAlgorithm.MLKem1024, kem.Algorithm);
+                Assert.Equal("CN=LAMPS WG, O=IETF", cert.Subject);
+                AssertExtensions.SequenceEqual(MLKemTestData.IncrementalSeed, kem.ExportPrivateSeed());
+                AssertExtensions.SequenceEqual(
+                    MLKemTestData.IetfMlKem1024PrivateKeyDecapsulationKey,
+                    kem.ExportDecapsulationKey());
+            }
+        }
+
+        [ConditionalTheory(nameof(MLKemIsNotSupported))]
+        [MemberData(nameof(StorageFlags))]
+        public static void ReadMLKem512PrivateKey_NotSupported(X509KeyStorageFlags keyStorageFlags)
+        {
+            byte[] pfxBytes = MLKemTestData.IetfMlKem512PrivateKeySeedPfx;
+            string pfxPassword = MLKemTestData.EncryptedPrivateKeyPassword;
+
+            // Windows when using the DefaultKeySet delays throwing no private key and instead acts as it the
+            // keyset does not exist. Exporting it again to PFX forces Windows to reconcile the fact the key
+            // didn't actually load.
+            if (PlatformDetection.IsWindows && keyStorageFlags == X509KeyStorageFlags.DefaultKeySet)
```

Comment:
I assume this is actually "when not using EphemeralKeySet"?  I wouldn't expect Exportable, MachineKeySet, etc to affect it.

Reply:
Yeah, but the `MemberData` only had DefaultKeySet and EphemeralKeySet. I flipped the condition to make more sense.